File: libs/parse/tests/fixtures/c/hello.c
#include <stdio.h>

/* Minimal C fixture used by libs/parse tests. The expected named-
 * node shape: translation_unit { preproc_include, comment,
 * function_definition { ... } }. */

int add(int a, int b) {
    return a + b;
}

int main(void) {
    int sum = add(40, 2);
    printf("%d\n", sum);
    return 0;
}

File: libs/parse/tests/fixtures/python/hello.py
"""Minimal Python fixture used by libs/parse tests.

Exercises module-level imports, dataclasses, type annotations,
generators, async functions, context managers, and f-strings — the
grammar paths most production Python code we'll scan goes through.
The triple-quoted docstring is the indentation-tracking external
scanner's first real test, since tree-sitter-python's scanner.c is
responsible for the INDENT/DEDENT tokens the grammar relies on.
"""

from __future__ import annotations

import asyncio
from dataclasses import dataclass
from typing import Generator


@dataclass
class Greeting:
    subject: str
    reps: int = 1

    def render(self) -> str:
        return f"Hello, {self.subject}!"


def add(a: int, b: int) -> int:
    return a + b


def counted(g: Greeting) -> Generator[str, None, None]:
    for _ in range(g.reps):
        yield g.render()


async def main() -> int:
    g = Greeting(subject="world", reps=add(40, 2))
    for line in counted(g):
        print(line)
    return 0


if __name__ == "__main__":
    asyncio.run(main())
